From ce9307bd06b31aab1b85383d4a9070c7fdb39498 Mon Sep 17 00:00:00 2001
From: Tejun Heo <htejun@gmail.com>
Date: Fri, 3 Aug 2007 02:22:22 +0900
Subject: [PATCH] libata-pmp-prep: implement EH fast-fail path
References: 288078

If PMP itself becomes inaccessible while trying to link a downstream
link, spending time to recover the downstream link doesn't make any
sense.  Make EH skip retry and fail fast if -ERESTART is received.

Signed-off-by: Tejun Heo <htejun@gmail.com>
---
 drivers/ata/libata-eh.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/ata/libata-eh.c	2007-10-31 17:44:41.000000000 -0400
+++ b/drivers/ata/libata-eh.c	2007-10-31 17:44:42.000000000 -0400
@@ -1977,7 +1977,7 @@ int ata_eh_reset(struct ata_link *link, 
 	if (rc == -EAGAIN)
 		rc = 0;
 
-	if (rc && try < ARRAY_SIZE(ata_eh_reset_timeouts)) {
+	if (rc && rc != -ERESTART && try < ARRAY_SIZE(ata_eh_reset_timeouts)) {
 		unsigned long now = jiffies;
 
 		if (time_before(now, deadline)) {
